### DISCUSSION TITLE
I need help with midi controller setup for multiple floppy drives

### DISCUSSION THREAD

**Discussion by Arg-xz** [SammyIAm/Moppy2#219] [Q&A]
Hi, I have 6 floppy drives connected to an arduino micro and a midi controller. Both are connected to a Raspberry Pi 4. 

The problem I have is simple, my midi controller works on midi channel 1 to my knowledge and allows me to play 1 of the six floppy drives. But it seems that the other floppy drives are on seperate channels because midi files play nicely on my floppies but I can only play one note at a time on one floppy drive. I've tried to change the mapper settings but that has resulted in me not play with my midi controller anything until I relaunch the program.

Could someone help me shove all six of my floppy drives to channel one or something so I could play multiple floppy drives at once on one midi controller?

**Comment by samphonic** [comment]
By default each channel is sent to a different drive.  You can use all the drives with one channel by changing the "Sub Address" script to the Round Robin option (I can't remember the exact string at the moment).  That will play each successive note on the next-higher-numbered drive.  You might need to adjust the script for 6 drives instead of 8.

**Reply by Arg-xz** [reply]
I did try that, but like I said earlier, when I change something in the script boxes, nothing plays afterwards, I'll try again tomorrow though.

**Reply by samphonic** [reply]
Take a screenshot of what you changed it to, just in case something's off.

**Reply by Arg-xz** [reply]
Will do

**Reply by Arg-xz** [reply]
Well, I didn't take a screen shot, but I have confirmed that
A: midi files play multiple floppy drives at once
B: midi controller plays one floppy drive at once
C: chancing the sub address box causes midi controller to not play any floppy drives until program relaunch (same applies to other boxes)

**Reply by samphonic** [reply]
Hrm. Is there anything in the logs? There should be a log file somewhere near the .bat, or in the terminal window (sorry I don't have anything set up to give you precise instructions).

A screenshot of after changing the sub address command would still be helpful.

**Comment by Arg-xz** [comment]
so I tried again things magically worked. it launched with the custom sub address option. Changing things still break it.

here's the log.
[New.txt](https://github.com/user-attachments/files/17091279/New.txt)

so things work sometimes and it's pretty jank, but I'll look in to it later, gotta sleep. I need to get a better source of power as well but it works at least a little.

**Reply by samphonic** [reply]
Hmm, it's possible there's some weirdness because of the many years of Java updates (the whole scripting engine got depreciated at some point).  What version of Java are you using?

I don't see anything that looks like an error in the logs.  A screenshot of the GUI might be helpful since I can't test/replicate here.

**Reply by Arg-xz** [reply]
I don't know what java version I have. It should be the default that comes with the raspbian I have installed? I'll get back to you later.
